### PR TITLE
feat: add dms rocketmq instances data source

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
  add dms rocketmq instances data source, use this data source to get the list of DMS RocketMQ instances.
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  add dms rocketmq instances data source
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/dms/' TESTARGS='-run TestAccDatasourceDmsRocketMQInstances_basic' 
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/dms/ -v -run TestAccDatasourceDmsRocketMQInstances_basic -timeout 360m -parallel 4 
=== RUN   TestAccDatasourceDmsRocketMQInstances_basic 
=== PAUSE TestAccDatasourceDmsRocketMQInstances_basic                                                            
=== CONT  TestAccDatasourceDmsRocketMQInstances_basic                                                            
--- PASS: TestAccDatasourceDmsRocketMQInstances_basic (681.73s)                                                  
PASS                                                                                                             
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dms       681.785s
```

### CODE REVIEW

File: docs/data-sources/dms_rocketmq_instances.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+The following arguments are supported:
+
+* `region` - (Optional, String) Specifies the region in which to query the data source.
+  If omitted, the provider-level region will be used.
+
+* `name` - (Optional, String) Specifies the name of the DMS RocketMQ instance.
+
+* `instance_id` - (Optional, String) Specifies the ID of the rocketMQ instance.
+
+* `status` - (Optional, String) Specifies the status of the DMS RocketMQ instance.
```
Comment (Lance52259) [inline docs/data-sources/dms_rocketmq_instances.md]: RocketMQ

File: docs/data-sources/dms_rocketmq_instances.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The resource ID.
+
+* `instances` - Indicates the list of DMS rocketMQ instances.
+  The [Instance](#DmsRocketMQInstances_Instance) structure is documented below.
+
+<a name="DmsRocketMQInstances_Instance"></a>
+The `Instance` block supports:
```
Comment (Lance52259) [inline docs/data-sources/dms_rocketmq_instances.md]:
```suggestion
* `instances` - Indicates the list of DMS rocketMQ instances.
  The [instances](#DmsRocketMQInstances_Instance) structure is documented below.

<a name="DmsRocketMQInstances_Instance"></a>
The `instances` block supports:

```

File: docs/data-sources/dms_rocketmq_instances.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+* `status` - Indicates the status of the DMS RocketMQ instance.
+
+* `description` - Indicates the description of the DMS RocketMQ instance.
+
+* `type` - Indicates the DMS RocketMQ instance type. Value: cluster.
+
+* `specification` - Indicates the instance specification. For a cluster DMS RocketMQ instance, VM specifications
+  and the number of nodes are returned.
+
+* `engine_version` - Indicates the version of the RocketMQ engine. Value: 4.8.0.
```
Comment (Lance52259) [inline docs/data-sources/dms_rocketmq_instances.md]:
```suggestion
* `engine_version` - Indicates the version of the RocketMQ engine.
```

File: docs/data-sources/dms_rocketmq_instances.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+<a name="DmsRocketMQInstances_Instance"></a>
+The `Instance` block supports:
+
+* `name` - Indicates the name of the DMS RocketMQ instance.
+
+* `status` - Indicates the status of the DMS RocketMQ instance.
+
+* `description` - Indicates the description of the DMS RocketMQ instance.
+
+* `type` - Indicates the DMS RocketMQ instance type. Value: cluster.
```
Comment (Lance52259) [inline docs/data-sources/dms_rocketmq_instances.md]:
```suggestion
* `type` - Indicates the DMS RocketMQ instance type.
```

File: docs/data-sources/dms_rocketmq_instances.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+* `availability_zones` - Indicates the list of availability zone names, where
+  instance brokers reside and which has available resources.
+
+* `maintain_begin` - Indicates the time at which the maintenance window starts. The format is HH:mm:ss.
+
+* `maintain_end` - Indicates the time at which the maintenance window ends. The format is HH:mm:ss.
+
+* `storage_space` - Indicates the message storage capacity, unit is GB.
+
+* `used_storage_space` - Indicates the used message storage space. Unit: GB.
```
Comment (Lance52259) [inline docs/data-sources/dms_rocketmq_instances.md]:
```suggestion
* `storage_space` - Indicates the message storage capacity, unit is GB.

* `used_storage_space` - Indicates the used message storage space, unit is GB.

```

File: docs/data-sources/dms_rocketmq_instances.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+* `broker_address` - Indicates the service data address.
+
+* `public_namesrv_address` - Indicates the public network metadata address.
+
+* `public_broker_address` - Indicates the public network service data address.
+
+* `resource_spec_code` - Indicates the resource specifications.
+
+<a name="DmsRocketMQInstances_InstanceCrossVpc"></a>
+The `InstanceCrossVpc` block supports:
```
Comment (Lance52259) [inline docs/data-sources/dms_rocketmq_instances.md]: ditto

File: huaweicloud/services/acceptance/dms/data_source_huaweicloud_rocketmq_instances_test.go [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+	rName := "data.huaweicloud_dms_rocketmq_instances.test"
+	dc := acceptance.InitDataSourceCheck(rName)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDatasourceDmsRocketMQInstances_config(name),
+			},
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/dms/data_source_huaweicloud_rocketmq_instances_test.go]: using dependency just ok.
```
data "huaweicloud_dms_rocketmq_instances" "test" {
  name = huaweicloud_dms_rocketmq_instance.test.name
}
```

File: docs/data-sources/dms_rocketmq_instances.md (hunk 1 above)
Comment (houpeng80) [reply docs/data-sources/dms_rocketmq_instances.md]: done

File: docs/data-sources/dms_rocketmq_instances.md (hunk 2 above)
Comment (houpeng80) [reply docs/data-sources/dms_rocketmq_instances.md]: done

File: docs/data-sources/dms_rocketmq_instances.md (hunk 3 above)
Comment (houpeng80) [reply docs/data-sources/dms_rocketmq_instances.md]: done

File: docs/data-sources/dms_rocketmq_instances.md (hunk 4 above)
Comment (houpeng80) [reply docs/data-sources/dms_rocketmq_instances.md]: done

File: docs/data-sources/dms_rocketmq_instances.md (hunk 5 above)
Comment (houpeng80) [reply docs/data-sources/dms_rocketmq_instances.md]: done

File: docs/data-sources/dms_rocketmq_instances.md (hunk 6 above)
Comment (houpeng80) [reply docs/data-sources/dms_rocketmq_instances.md]: done

File: huaweicloud/services/acceptance/dms/data_source_huaweicloud_rocketmq_instances_test.go (hunk 7 above)
Comment (houpeng80) [reply huaweicloud/services/acceptance/dms/data_source_huaweicloud_rocketmq_instances_test.go]: done

File: docs/data-sources/dms_rocketmq_instances.md [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+* `availability_zones` - Indicates the list of availability zone names, where
+  instance brokers reside and which has available resources.
+
+* `maintain_begin` - Indicates the time at which the maintenance window starts. The format is HH:mm:ss.
+
+* `maintain_end` - Indicates the time at which the maintenance window ends. The format is HH:mm:ss.
+
+* `storage_space` - Indicates the message storage capacity, unit is GB.
+
+* `used_storage_space` - Indicates the used message storage space, Unit: GB.
```
Comment (Lance52259) [inline docs/data-sources/dms_rocketmq_instances.md]: keep the descriptions uniform.

File: huaweicloud/services/dms/data_source_huaweicloud_dms_rocketmq_instances.go [hunk 9]
```diff
@@ -0,0 +1,404 @@
...
+			},
+			"specification": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Indicates the instance specification.`,
+			},
+			"engine_version": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Indicates the version of the RocketMQ engine. Value: 4.8.0.`,
```
Comment (Lance52259) [inline huaweicloud/services/dms/data_source_huaweicloud_dms_rocketmq_instances.go]:
```suggestion
				Description: `Indicates the version of the RocketMQ engine.`,
```

File: huaweicloud/services/dms/data_source_huaweicloud_dms_rocketmq_instances.go [hunk 10]
```diff
@@ -0,0 +1,404 @@
...
+			},
+			"storage_space": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: `Indicates the message storage capacity, unit is GB.`,
+			},
+			"used_storage_space": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: `Indicates the used message storage space. Unit: GB.`,
```
Comment (Lance52259) [inline huaweicloud/services/dms/data_source_huaweicloud_dms_rocketmq_instances.go]: ditto.

File: docs/data-sources/dms_rocketmq_instances.md (hunk 8 above)
Comment (houpeng80) [reply docs/data-sources/dms_rocketmq_instances.md]: done

File: huaweicloud/services/dms/data_source_huaweicloud_dms_rocketmq_instances.go (hunk 10 above)
Comment (houpeng80) [reply huaweicloud/services/dms/data_source_huaweicloud_dms_rocketmq_instances.go]: done

File: huaweicloud/services/dms/data_source_huaweicloud_dms_rocketmq_instances.go (hunk 9 above)
Comment (houpeng80) [reply huaweicloud/services/dms/data_source_huaweicloud_dms_rocketmq_instances.go]: done

File: docs/data-sources/dms_rocketmq_instances.md [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+  If omitted, the provider-level region will be used.
+
+* `name` - (Optional, String) Specifies the name of the DMS RocketMQ instance.
+
+* `instance_id` - (Optional, String) Specifies the ID of the RocketMQ instance.
+
+* `status` - (Optional, String) Specifies the status of the DMS RocketMQ instance.
+
+* `exact_match_name` - (Optional, String) Specifies whether to search for the instance that precisely matches a
+  specified instance name. Value options: **true**, **false**.
```
Comment (Lance52259) [inline docs/data-sources/dms_rocketmq_instances.md]: which default？

File: docs/data-sources/dms_rocketmq_instances.md [hunk 12]
```diff
@@ -0,0 +1,121 @@
...
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The resource ID.
+
+* `instances` - Indicates the list of DMS RocketMQ instances.
+  The [Instance](#DmsRocketMQInstances_Instance) structure is documented below.
+
+<a name="DmsRocketMQInstances_Instance"></a>
+The `instances` block supports:
```
Comment (Lance52259) [inline docs/data-sources/dms_rocketmq_instances.md:43]: the attribute `id` is missing.

File: huaweicloud/services/dms/data_source_huaweicloud_dms_rocketmq_instances.go [hunk 13]
```diff
@@ -0,0 +1,405 @@
...
+		crossVpcInfo := utils.PathSearch("cross_vpc_info", v, nil)
+		var crossVpcAccess []map[string]interface{}
+		if crossVpcInfo != nil {
+			crossVpcAccess, err = flattenConnectPorts(crossVpcInfo.(string))
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		rst = append(rst, map[string]interface{}{
```
Comment (Lance52259) [inline huaweicloud/services/dms/data_source_huaweicloud_dms_rocketmq_instances.go:349]: please append `id`

File: docs/data-sources/dms_rocketmq_instances.md (hunk 11 above)
Comment (houpeng80) [reply docs/data-sources/dms_rocketmq_instances.md]: done

File: docs/data-sources/dms_rocketmq_instances.md (hunk 12 above)
Comment (houpeng80) [reply docs/data-sources/dms_rocketmq_instances.md:43]: done

File: huaweicloud/services/dms/data_source_huaweicloud_dms_rocketmq_instances.go (hunk 13 above)
Comment (houpeng80) [reply huaweicloud/services/dms/data_source_huaweicloud_dms_rocketmq_instances.go:349]: done

File: docs/data-sources/dms_rocketmq_instances.md [hunk 14]
```diff
@@ -0,0 +1,123 @@
...
+  If omitted, the provider-level region will be used.
+
+* `name` - (Optional, String) Specifies the name of the DMS RocketMQ instance.
+
+* `instance_id` - (Optional, String) Specifies the ID of the RocketMQ instance.
+
+* `status` - (Optional, String) Specifies the status of the DMS RocketMQ instance.
+
+* `exact_match_name` - (Optional, String) Specifies whether to search for the instance that precisely matches a
+  specified instance name. Value options: **true**, **false**. Default to **false**.
```
Comment (Lance52259) [inline docs/data-sources/dms_rocketmq_instances.md]:
```suggestion
  specified instance name. Value options: **true**, **false**. Defaults to **false**.

```

Comment (houpeng80) [reply docs/data-sources/dms_rocketmq_instances.md]: done

**Pull request merged**